Source/Core/DolphinQt2/CheatsManager.cpp

```diff
@@ -0,0 +1,340 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/CheatsManager.h"
+
+#include <QComboBox>
+#include <QDialogButtonBox>
+#include <QGroupBox>
+#include <QLabel>
+#include <QLineEdit>
+#include <QPushButton>
+#include <QRadioButton>
+#include <QSplitter>
+#include <QTabWidget>
+#include <QTableWidget>
+#include <QVBoxLayout>
+
+#include "Core/ConfigManager.h"
+#include "Core/HW/Memmap.h"
+#include "Core/PowerPC/PowerPC.h"
+#include "UICommon/GameFile.h"
+
+#include "DolphinQt2/Config/ARCodeWidget.h"
+#include "DolphinQt2/Config/GeckoCodeWidget.h"
+#include "DolphinQt2/GameList/GameListModel.h"
+#include "DolphinQt2/Settings.h"
+
+constexpr int TYPE_BYTE_INDEX = 0;
+constexpr int TYPE_SHORT_INDEX = 1;
+constexpr int TYPE_INT_INDEX = 2;
+constexpr int TYPE_FLOAT_INDEX = 3;
+constexpr int TYPE_DOUBLE_INDEX = 4;
+constexpr int TYPE_STRING_INDEX = 5;
+
+constexpr int CMP_EQUAL_INDEX = 0;
```

Comment:
This can likely be an enum class, where the currentIndex is just casted once instead.